src/hdlc.c

```diff
@@ -59,53 +70,80 @@ static uint16_t fcs_tab[] = {
 };
 
 /*
- * 16-bit frame check sequence from RFC1662.
+ * Calculates a new FCS from the current FCS and new data.
+ *
+ * @param[in] sum      Current FCS value.
+ * @param[in] seq      Array of new data.
+ * @param[in] length   Length of the array of new data.
+ * @return             new FCS value.
  */
-static uint16_t frame_checksum_16bit(uint16_t sum, uint8_t *seq, size_t length)
+static uint16_t frame_checksum_16bit(uint16_t sum, const uint8_t *seq, size_t length)
```

Comment:
is this `const` ok here? there is `*seq++` on the new line 83

Reply:
If it weren't const the compiler would complain. We don't assign to `*seq++` so that's OK.

Reply:
oh, yes, you are right. I've found a good write-up about this topic: https://www.geeksforgeeks.org/difference-const-char-p-char-const-p-const-char-const-p/